### DISCUSSION TITLE
createQueryBuilder doesn't work using leftJoin, limit and orderBy persist false field

### DISCUSSION THREAD

**Discussion by idanavr** [mikro-orm/mikro-orm#4616] [Q&A]
**Describe the bug**
When combining `leftJoinAndSelect`, `orderBy` (persist: false) and `limit`, while using createQueryBuilder it causes error:
"order by "persistFalseField" desc - cross-database references are not implemented"

**Stack trace**
```
select "table1"."id", "abc" as persistFalseField, "table2s"."id" as "table2s__id", "table2s"."related_id" as "table2s__related_id" 
from "table1" as "table1" 
left join "table2" as "table2s" on "table1"."id" = "table2s"."related_id" 
where "table1"."id" in (
  select "table1"."id" from (
    select "table1"."id", """abc""" as "persistFalseField" 
    from "table1" as "table1" 
    left join "table2" as "table2s" on "table1"."id" = "table2s"."related_id" 
    group by "table1"."id" order by min("persistFalseField") desc limit 200
  ) as "table1"
) order by "persistFalseField" desc - column "abc" does not exist
```

**To Reproduce**
Make a query using these 3 functions, while orderBy with `persist: false` field.

**Expected behavior**
I would expect it to work.
Not sure why the query builder need to do "group by" when using `limit`, while it can just do the orderBy and limit without it.

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | 18.15.0 |
| typescript | 5.1.6 |
| mikro-orm | 5.7.14 |
| your-driver | Postgres |

**Comment by B4nan** [comment]
I need to see a complete reproduction for such issues, I can't debug a generated SQL query. 

> Not sure why the query builder need to do "group by" when using limit, while it can just do the orderBy and limit without it.

Because you are joining to-many relation, and doing that results in a cartesian product in the final results - you would be limiting that instead of the number of the root entities (and I am sure that is what you actually want).

**Reply by bobgubko** [reply]
But what if I want to select root entities? Is there a way to achieve it? In my case I manually add second field of to-many relation

```
const post = await qb
  .select('p.*')
  .join('p.postEmail', 'pe', {
    'pe.userId': 1,
  })
  .limit(1)
  .getSingleResult()
```

but generated SQL is not correct, `pe.userId` is missing in subquery. Is it a bug or I'm missing something here?

```
select
  `p`.*
from
  `records` as `p`
  inner join `post_emails` as `pe` on `p`.`id` = `pe`.`postId`
  and `pe`.`userId` = 1
where
  `p`.`id` in (
    select
      `p`.`id`
    from
      (
        select
          `p`.`id`
        from
          `records` as `p`
          inner join `post_emails` as `pe` on `p`.`id` = `pe`.`postId`
        group by
          `p`.`id`
        limit
          1
      ) as `p`
  )
```

**Reply by B4nan** [reply]
You need to use `joinAndSelect()` if you want to have that relation populated:

```ts
const post = await qb
//  .select('p.*') // this is actually implicit
  .joinAndSelect('p.postEmail', 'pe', {
    'pe.userId': 1,
  })
  .limit(1)
  .getSingleResult()
```
